### PR TITLE
Preparation for V3 release

### PR DESCRIPTION
**Pull request by milkshakeuk**
- update nuspec to be more correct in terms of dependencies
- update csproj to support individual NuGet packages
- update dotnet SDK to 5.0.301 + GitHub actions

### CODE REVIEW

File: build/nHapi.v3.nuspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
     <metadata>
         <id>nHapi</id>
-        <version>3.0.0-preview.3</version>
+        <version>3.0.0.0</version>
```
Comment (AMCN41R) [inline build/nHapi.v3.nuspec]: should this just be `3.0.0`?

File: build/nHapi.v3.nuspec [hunk 2]
```diff
@@ -13,21 +13,22 @@
 NHapi allows Microsoft .NET developers to easily use an HL7 2.x object model. This object model allows for parsing and encoding HL7 2.x data to/from Pipe Delimited or XML formats. A very handy program for use in the health care industry.
 
 This project is NOT affiliated with the HL7 organization. This software just conforms to the HL7 2.x specifications.</description>
-        <releaseNotes>https://github.com/nHapiNET/nHapi/releases/tag/v3.0.0-preview.3</releaseNotes>
+        <releaseNotes>https://github.com/nHapiNET/nHapi/releases/tag/v3.0.0.0</releaseNotes>
```
Comment (AMCN41R) [inline build/nHapi.v3.nuspec]: and here?

**Review by AMCN41R — APPROVED** [review]

File: build/nHapi.v3.nuspec (hunk 1 above)
Comment (milkshakeuk) [reply build/nHapi.v3.nuspec]: already done

File: build/nHapi.v3.nuspec (hunk 2 above)
Comment (milkshakeuk) [reply build/nHapi.v3.nuspec]: already changed

**Pull request merged**